tests/sharing/tests/multiple_sharings.rb

```diff
@@ -0,0 +1,107 @@
+    file = "../fixtures/wet-cozy_20160910__©M4Dz.jpg"
+    opts = CozyFile.options_from_fixture(file, dir_id: folder.couch_id)
+    file = CozyFile.create inst_alice, opts
+
+    sharing = Sharing.new
+    sharing.rules << Rule.sync(folder)
+    sharing.members << inst_alice << contact_bob
+    sharing.members << contact_charlie
+    inst_alice.register sharing
+
+    # Accept the sharing
+    sleep 1
+    inst_bob.accept sharing
+    inst_charlie.accept sharing
+
+    sleep 5
+
+    # Check the folders are the same
+    path = CGI.escape "/#{Helpers::SHARED_WITH_ME}/#{folder.name}"
+    folder_bob = Folder.find_by_path inst_bob, path
+    assert_equal folder_bob.name, folder.name
+    folder_charlie = Folder.find_by_path inst_charlie, path
+    assert_equal folder_charlie.name, folder.name
+
+    #  Share a folder with Bob and Charlie, in 2 different sharings
+    folder = Folder.create inst_alice
+    child1 = Folder.create inst_alice, {dir_id: folder.couch_id}
+    sharing = Sharing.new
+    sharing.rules << Rule.sync(folder)
+    sharing.members << inst_alice << contact_bob
+    inst_alice.register sharing
+    sleep 1
+    inst_bob.accept sharing
+    sharing.members = []
+    sharing.members << inst_alice << contact_bob
+    inst_alice.register sharing
+    sleep 1
+    inst_charlie.accept sharing
+
+    sleep 5
+
+    path = CGI.escape "/#{Helpers::SHARED_WITH_ME}/#{folder.name}/#{child1.name}"
+    child1_bob = Folder.find_by_path inst_bob, path
+    assert_equal child1_bob.name, child1.name
+    child1_charlie = Folder.find_by_path inst_charlie, path
+    assert_equal child1_charlie.name, child1.name
+
+    # Propagate a change from Alice's side
+    new_name = Faker::Internet.slug
+    child1.rename inst_alice, new_name
+    sleep 10
+    path = CGI.escape "/#{Helpers::SHARED_WITH_ME}/#{folder.name}/#{new_name}"
+    child1_bob = Folder.find_by_path inst_bob, path
+    assert_equal child1_bob.name, new_name
+    child1_charlie = Folder.find_by_path inst_charlie, path
+    assert_equal child1_charlie.name, new_name
+
+    # Propagate a change from Bob's side
+    new_name = Faker::Internet.slug
+    child1_bob.rename inst_bob, new_name
+    sleep 10
+    path = CGI.escape "/#{Helpers::SHARED_WITH_ME}/#{folder.name}/#{new_name}"
+    child1_alice = Folder.find inst_alice, child1.couch_id
+    assert_equal child1_alice.name, new_name
+    child1_charlie = Folder.find_by_path inst_charlie, path
+    assert_equal child1_charlie.name, new_name
+
+    # Check that the files are the same on disk
+    da = File.join Helpers.current_dir, inst_alice.domain, folder.name
+    db = File.join Helpers.current_dir, inst_bob.domain,
+                   Helpers::SHARED_WITH_ME, sharing.rules.first.title
+    dc = File.join Helpers.current_dir, inst_charlie.domain,
+                   Helpers::SHARED_WITH_ME, sharing.rules.first.title
+    diff = Helpers.fsdiff da, db
+    diff.must_be_empty
+    diff = Helpers.fsdiff da, dc
+    diff.must_be_empty
+
```

Comment:
This scenario looks good to me. Can you also try to add a file in the shared folder?